### DISCUSSION TITLE
AWS S3 Signed url support for grouped requests

### DISCUSSION THREAD

**Discussion by jiry1994** [rpldy/react-uploady#732] [Q&A]
Is it possible / idea to use grouped request for requesting multiple signed urls at once. And use the useRequestPreSend for returning an array of of options object containing the multiple signed urls. It seems like its now only possible to handle 1 file by 1.

In this way you can request for example 3 signed urls at once instead of 1 by 1.

Or is there another way to achieve this?

**I tried the following**
```
        <Uploady
            destination={{}}
            grouped={true}
            maxGroupSize={3}
            autoUpload={true}
            concurrent={true}
            maxConcurrent={3}
            enhancer={retryEnhancer}
            accept='image/png, image/jpeg, image/gif'>
            <Flex vertical={true} gap='large' className='h-full'>
                <UploadDragDrop />
                <UploadBatchLayout />
            </Flex>
        </Uploady>
```

```
    useRequestPreSend(async (batch) => {
        try {
            const batchItems = batch.items;

            const photoResults = await getSignedUploadUrls(photosetId, { batchItems, duplicateOption }, dispatch);

            for (const photoResult of photoResults) {
                const { uploadUrl, photo, skipped, replaced } = photoResult;

                const fileItem = batchItems.find(({ file }) => file.name === photo.name);

                fileItem.file.skipped = skipped;
                fileItem.file.replaced = replaced;

                if (skipped) {
                    return false;
                }

                return {
                    options: {
                        sendWithFormData: false,
                        destination: {
                            method: 'PUT',
                            url: uploadUrl,
                            headers: { 'Content-Type': fileItem.file.type },
                        },
                    },
                };
            }
        } catch (error) {
           // handle error
        }
    });
```

**Comment by yoavniran** [comment]
moving to Discussions. will answer there

**Comment by yoavniran** [comment]
hi @jiry1994 
By design, if you're using _grouped: true_ it means the files are sent together as part of a single request, and therefore, a single URL.

I think what you're after is _concurrent_(parallel) uploading. You can turn it on using the _concurrent_ prop together with the _maxConcurrent_ prop.

Then, you can use [withBatchStartUpdate](https://react-uploady.org/docs/api/hocs/withBatchStartUpdate/), in a very similar fashion to what's described in this [thread](https://github.com/rpldy/react-uploady/discussions/708#discussioncomment-10196616).

Hope this helps.
